src/CoreOperations/ModRegistry/__init__.py

```diff
@@ -121,7 +121,15 @@ def unregister_mod(self, index):
             self.ui.log(translate("ModRegistry", "The following error occured when trying to delete {mod_name}: {error}").format(mod_name=mod_name, error=e))
         finally:
             self.update_mods()
-            
+
+    def open_mod_folder(self, index):
+        full_path = self.profile_manager.mods[index].path
```

Comment:
I would add an extra safety check in here with `os.path.isdir()` just to make sure that the path is definitely a folder, just in case something goes catastrophically wrong in another part of the mod manager. E.g.,
```python
if not os.path.isdir(full_path):
    raise ValueError(f"'{full_path}' is not a directory")
```

Reply:
Great! I thought it will bet taken care by Try/Except block!!

Reply:
If the directory doesn't exist it will be dealt with in the try/except block - my thought was that if the mod folder somehow points to a file instead of a directory, `QtGui.QDesktopServices.openUrl` would tell the OS to open that file, which would be really unexpected. I didn't consider the directory simply not existing, so maybe we should instead do

```python
if os.path.exists(full_path) and not os.path.isdir(full_path):
    self.ui.log(translate("ModRegistry", "CRITICAL INTERNAL ERROR: '{mod_path}' is a file and not a directory. Please report this as a bug, with instructions on how to reproduce.".format(mod_path=full_path)))
    return
if not os.path.exists(full_path):
    self.ui.log(translate("ModRegistry", "CRITICAL INTERNAL ERROR: '{mod_path}' does not exist and cannot be opened. Please report this as a bug, with instructions on how to reproduce.".format(mod_path=full_path)))
    return
```

The second case will be caught by the try/except as you correctly said, but it's also a good idea to communicate to the user the most information possible on why something isn't working, and have a generic error message as a fallback in case there are errors that we did not forsee. If they end up being common problems we can then add additional specific error-checks to make those situations more useful/eliminate them if they're bugs :)